golang/cosmos/x/vbank/vbank.go

```diff
@@ -69,6 +69,26 @@ func marshalBalanceUpdate(addressToBalance map[string]sdk.Coins) ([]byte, error)
 	return json.Marshal(&event)
 }
 
+// rewardRate calculates the rate for dispensing the pool of coins over
+// the specified number of blocks. Fractions are rounded up. In other
+// words, it returns the smallest Coins such that pool is exhausted
+// after #blocks withdrawals.
+func rewardRate(pool sdk.Coins, blocks int64) sdk.Coins {
+	coins := make([]sdk.Coin, 0)
+	if blocks > 0 {
+		for _, coin := range pool {
+			amt := coin.Amount.Int64()
```

Comment:
Just to be pedantic, please don't cast a `coin.Amount` to an `Int64`, since it technically could overflow.  Please use `sdk.Int` comparisons and math instead.

This may seem silly at this point, but if we ever decide to use coins which are represented in about 18 decimalPlaces (such as most coins transferred from Ethereum), we quickly overflow `Int64`.

Reply:
Done. Checked that this was the only int64 assumption and added a test with large amounts.